libdirections-hybrid/src/main/java/com/mapbox/navigation/route/hybrid/MapboxHybridRouter.kt

```diff
@@ -20,37 +21,40 @@ class MapboxHybridRouter(
 
     private val networkStatusService = NetworkStatusService(context)
     private val jobControl = ThreadController.getIOScopeAndRootJob()
-    private val offBoardRouterHandler: RouterHandler by lazy {
+    private val offboardRouterHandler: RouterHandler by lazy {
         RouterHandler(mainRouter = offboardRouter, reserveRouter = onboardRouter)
     }
-    private val onBoardRouterHandler: RouterHandler by lazy {
+    private val onboardRouterHandler: RouterHandler by lazy {
         RouterHandler(mainRouter = onboardRouter, reserveRouter = offboardRouter)
     }
 
+    // internal for testing
```

Comment:
Instead of exposing this, couldn't we rely on `public` APIs or mock other dependencies somehow to get the desired behavior on the test?

Reply:
we don't have access to needed properties via public API and we can't mock them. 
`NetworkStatusService ` is private
We need somehow control `channel` result. That's why I expose the same function as channel use to notify onNetworkStateUpdate